src/query.c

```diff
@@ -920,7 +946,179 @@ void QueryResult_Free(QueryResult *q) {
   free(q);
 }
 
-int QueryResult_Serialize(QueryResult *r, RedisSearchCtx *sctx, RSSearchRequest *req) {
+// Strip spaces from a buffer in place.
+static size_t stripDuplicateSpaces(char *s, size_t n) {
+  int isLastSpace = 0;
+  size_t oix = 0;
+  char *out = s;
+  for (size_t ii = 0; ii < n; ++ii) {
+    if (isspace(s[ii])) {
+      if (isLastSpace) {
+        continue;
+      } else {
+        isLastSpace = 1;
+        out[oix++] = ' ';
+      }
+    } else {
+      isLastSpace = 0;
+      out[oix++] = s[ii];
+    }
+  }
+  return oix;
+}
+
+#define ELLIPSIS "... "
+
+static void sendSummarizedField(const Query *q, const RSSearchRequest *req, RedisModuleCtx *ctx,
+                                const ReturnedField *fieldInfo, const RedisModuleString *text) {
+  FragmentList frags;
+  const FragmentTerm *terms = (const FragmentTerm *)Buffer_GetData(&q->termsMeta);
+  size_t numTerms = BUFFER_GETSIZE_AS(&q->termsMeta, FragmentTerm);
+
+  // Dump the terms:
+  // for (size_t ii = 0; ii < numTerms; ++ii) {
+  //   printf("Have term '%.*s'\n", (int)terms[ii].len, terms[ii].tok);
+  // }
+
+  Stemmer *stemmer = NewStemmer(SnowballStemmer, q->language);
```

Comment:
Can we avoid stemming in the highlighter? It's really slow. Doesn't the new offset based impl. allow removing this?